src/connector/src/common.rs

```diff
@@ -604,13 +608,28 @@ impl NatsCommon {
     > {
         let context = self.build_context().await?;
         let stream = self.build_or_get_stream(context.clone(), stream).await?;
-        let subject_name = self
-            .subject
-            .replace(',', "-")
-            .replace(['.', '>', '*', ' ', '\t'], "_");
+        let subject_name: String = match self.filter_subject {
```

Comment:
Why do you use `filter_subject` directly as the subject name? Is it as expected?